mono/mini/aot-compiler.c

```diff
@@ -10649,7 +11019,16 @@ mono_compile_assembly (MonoAssembly *ass, guint32 opts, const char *aot_options)
 		}
 	}
 
-	load_profile_files (acfg);
+	if (acfg->aot_opts.profile_file)
+		load_profile_file (acfg, acfg->aot_opts.profile_file);
+
+	{
+		int method_index;
```

Comment:
Fair enough, one wait this will be improved is when we support manipulating multiple files, so we'd be able to merge multiple profile runs.

For example, say we merge the profile files of running csc on corlib and System.Core and that gives us much better coverage over the whole BCL than just one of them.